rust/actyx/ax-core/src/private_key.rs

```diff
@@ -136,6 +157,54 @@ pub fn load_dev_cert(path: Option<PathBuf>) -> ActyxOSResult<DeveloperCertificat
     serde_json::from_str(&s).ax_err_ctx(ActyxOSCode::ERR_INVALID_INPUT, "reading developer certificate")
 }
 
+#[derive(Debug, Clone)]
+/// Newtype wrapper around a path to key material, to be used with
+/// structopt/clap.
+pub struct KeyPathWrapper(PathBuf);
+
+impl FromStr for KeyPathWrapper {
+    type Err = ActyxOSError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self(s.into()))
+    }
+}
+
+impl fmt::Display for KeyPathWrapper {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0.display())
+    }
+}
+
+// NOTE(duarte): there has to be a better way of doing this
+impl TryFrom<&Option<KeyPathWrapper>> for AxPrivateKey {
+    type Error = ActyxOSError;
+    fn try_from(k: &Option<KeyPathWrapper>) -> Result<Self, Self::Error> {
+        if let Some(path) = k {
+            path.0
+                .to_str()
+                .and_then(|s| s.parse::<AxPrivateKey>().ok())
+                .ok_or(ActyxOSError::internal(""))
```

Comment:
not sure I recall where this is used, but it feels suspicious that we erase the error message and replace it by a meaningless one